test/e2e/common/node/runtimeclass.go

```diff
@@ -375,12 +374,15 @@ func createRuntimeClass(ctx context.Context, f *framework.Framework, name, handl
 	return rc.GetName()
 }
 
-func expectPodRejection(ctx context.Context, f *framework.Framework, pod *v1.Pod) {
-	_, err := f.ClientSet.CoreV1().Pods(f.Namespace.Name).Create(ctx, pod, metav1.CreateOptions{})
-	gomega.Expect(err).To(gomega.HaveOccurred(), "should be forbidden")
-	if !apierrors.IsForbidden(err) {
-		framework.Failf("expected forbidden error, got %#v", err)
-	}
+func expectPodRejection(ctx context.Context, f *framework.Framework, rcName string) {
+	gomega.Eventually(ctx, func() error {
+		pod := e2eruntimeclass.NewRuntimeClassPod(rcName)
+		_, err := f.ClientSet.CoreV1().Pods(f.Namespace.Name).Create(ctx, pod, metav1.CreateOptions{DryRun: []string{metav1.DryRunAll}})
+		if !apierrors.IsForbidden(err) {
+			return err
```

Comment:
If err is nil then this will return nil, and be marked as success. I think you need a add a `if err == nil` check before this.